File: dotnet/cafdemalihapa/Applications/DirectoryManagement/DirectoryOpenStrategies/WindowsDirectoryOpenStrategy.cs
using System.Runtime.InteropServices;
using Microsoft.Extensions.Logging;

namespace cafdemalihapa.Applications.DirectoryManagement.DirectoryOpenStrategies
{
    public class WindowsDirectoryOpenStrategy(ILogger<WindowsDirectoryOpenStrategy> logger)
        : DirectoryOpenStrategy(logger)
    {
        protected override string FileName => "explorer.exe";

        protected override string TransformPath(string path) =>
            path.NormalizeSlashes(PathUtility.SlashStyle.ForceBackslash);

        public override bool CanHandle() => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/DirectoryManagement/DirectoryOpenStrategies/WindowsDirectoryOpenStrategy.fs
namespace cafdemwimu.console.Applications.DirectoryManagement.DirectoryOpenStrategies

open System.Runtime.InteropServices
open Microsoft.Extensions.Logging
open cafdemwimu.console.Applications

type WindowsDirectoryOpenStrategy(logger: ILogger<WindowsDirectoryOpenStrategy>) =
    inherit DirectoryOpenStrategy(logger)

    override _.FileName = "explorer.exe"

    override _.TransformPath(path) =
        PathUtility.NormalizeSlashes(path, PathUtility.SlashStyle.ForceBackslash)

    override _.CanHandle() = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
